demos/dotnet/dotnet/NugetAction.cs

```
@@ -0,0 +1,188 @@
+                file.WriteLine(@"        ""dnxcore50"": { }");
+                file.WriteLine(@"    }");
+                file.WriteLine(@"}");
+
+                //"runtimes": {
+                //"win7-x86": { },
+                //"win7-x64": { }
+                //},
+            }
+        }
+
+        private static void CreateNugetConfig(ProjectProperties properties)
+        {
+            var fileName = Path.Combine(properties.ToolsDirectory, "nuget.config");
+            var fs = new FileStream(fileName, FileMode.Create);
+            using (var file = new StreamWriter(fs, Encoding.UTF8))
+            {
+                file.WriteLine(@"<?xml version = ""1.0"" encoding=""utf-8""?>");
+                file.WriteLine(@"<configuration>");
+                file.WriteLine(@"    <packageSources>");
+
+                file.WriteLine(
+                    @"        <add key = ""netcore-prototype"" value=""https://www.myget.org/F/netcore-package-prototyping""/>");
+                file.WriteLine(
+                    @"        <add key = ""nuget.org"" value = ""https://api.nuget.org/v3/index.json"" protocolVersion = ""3""/>");
+                file.WriteLine(@"        <add key = ""nuget.org"" value = ""https://www.nuget.org/api/v2/""/>");
+
+                file.WriteLine(@"    </packageSources>");
+                file.WriteLine(@"</configuration>");
+            }
+        }
+
+        public static bool RestorePackagesAction(ProjectProperties properties, Log log, string nugetFile, string jsonFile)
+        {
+            Console.WriteLine("restoring packages");
+
+            if (!File.Exists(nugetFile))
+            {
+                Console.WriteLine("Could not find file {0}.", nugetFile);
+                return false;
+            }
+
+            if (!File.Exists(jsonFile))
+            {
+                Console.WriteLine("Could not find file {0}.", jsonFile);
+                return false;
+            }
+
```

For all of the above, it seems we'd want to throw an Argument Exception, no?